### ISSUE TITLE
Nodes: Location data for nested object literal incorrect.

### ISSUE THREAD

**Issue by edwinvanrijkom** [jashkenas/coffeescript#4205]
```js
> require("coffee-script").nodes("x:\n\ty:1").expressions[0].base.properties[0]
{ variable: 
   { base: { value: 'x', locationData: [Object] },
     properties: [],
     locationData: { first_line: 0, last_line: 0, first_column: 0, last_column: 0 } },
  value: 
   { base: 
      { generated: true,
        properties: [Object],
        objects: [Object],
        locationData: [Object] },
     properties: [],
     locationData: { first_line: 1, first_column: 1, last_line: 1, last_column: 3 } },
  context: 'object',
  param: undefined,
  subpattern: undefined,
  operatorToken: 
   { value: ':',
     locationData: { first_line: 0, first_column: 1, last_line: 0, last_column: 1 } },
  locationData: { first_line: 0, first_column: 0, last_line: 1, last_column: 4 } }
```

Note that `last-column` on the second line says `4` whereas it should read `3` (there are no more than 3 characters on the last line). The bad location data seems to propagate up to the parent `Assign` and `Object` and `Body` nodes.

When not nested, the column gets reported correctly:

```js
> require("coffee-script").nodes("y:1").expressions[0]
{ base: 
   { generated: true,
     properties: [ [Object] ],
     objects: [ [Object] ],
     locationData: { first_line: 0, first_column: 0, last_line: 0, last_column: 2 } },
  properties: [],
  locationData: { first_line: 0, first_column: 0, last_line: 0, last_column: 2 } }
```

Possibly related to https://github.com/jashkenas/coffeescript/issues/4114.

**Comment by koenbok** [comment]
@jashkenas @michaelficarra are we overlooking something? If not we can see if we can contribute a fix.

**Comment by vendethiel** [comment]
Isn't the `\t` character simply counted here?

**Comment by GeoffreyBooth** [comment]
This got fixed sometime before 1.12.7, probably thanks to one of @alangpierce’s pull requests:
```js
> require("coffeescript").VERSION
'1.12.7'
> require("coffeescript").nodes("x:\n\ty:1").expressions[0].base.properties[0]
Assign {
  variable:
   Value {
     base: PropertyName { value: 'x', locationData: [Object] },
     properties: [],
     locationData: { first_line: 0, last_line: 0, first_column: 0, last_column: 0 } },
  value:
   Value {
     base:
      Obj {
        generated: true,
        properties: [Array],
        objects: [Array],
        locationData: [Object] },
     properties: [],
     locationData: { first_line: 1, first_column: 1, last_line: 1, last_column: 3 } },
  context: 'object',
  param: undefined,
  subpattern: undefined,
  operatorToken:
   Literal {
     value: ':',
     locationData: { first_line: 0, first_column: 1, last_line: 0, last_column: 1 } },
  moduleDeclaration: undefined,
  locationData: { first_line: 0, first_column: 0, last_line: 1, last_column: 3 } }
```
